### DISCUSSION TITLE
The lazer tournament client shouldn't automatically assign flags based on a team's acronyms

### DISCUSSION THREAD

**Discussion by 0x84f** [ppy/osu#17539] [Q&A]
When making a team in the lazer tournament client, if you add a acronym it will automatically search for a country flag to add, when you remove the flag it will only stay removed until you reboot the client, at which point the flags will be back and they need to be removed.
[Example video](https://streamable.com/wa0wu9)

**Comment by peppy** [comment]
Can you explain what the issue here is? It's not perfect behaviour, but this shouldn't cause any actual issues right?

**Reply by 0x84f** [reply]
No, it doesn't cause any actual issues however it's annoying to constantly have to remove flags from the team names and if you don't it usually causes confusion to the viewers as to why a team has a completely random country flag.

**Comment by bdach** [comment]
i've just tested and this seems to work correctly here. i've removed the flag name from a team, saved changes, and then restarted game. the flag name didn't return.

@twoonetwoseven can you link an example `bracket.json` to test with?

**Reply by 0x84f** [reply]
[Here](https://2127.s-ul.eu/X0nBXKZ2.json)

**Comment by bdach** [comment]
this is happening because of this logic:

https://github.com/ppy/osu/blob/0020102810c8c3c2c4b8a319781c0f2ceb74dd48/osu.Game.Tournament/Models/TournamentTeam.cs#L64-L76

json deserialisation is triggering these paths on tournament client launch, and these `ValueChanged` subscriptions fire with `OldValue` being an empty string, and `NewValue` being the value read from json. because any string starts with an empty string, this logic fires wrongly and starts overwriting things.

**Reply by peppy** [reply]
~~Seems like a simple null/length check will fix this without too much further consideration. Will apply that for now.~~

This is a bit harder to handle correctly, as that logic *intends* to do precisely what the user here is looking to not do.

@twoonetwoseven can you explain in the first place why you don't want to use acronyms? is it because in your tournament you don't have the concept of acronyms?

**Reply by 0x84f** [reply]
The tournament host did not ask for teams to provide acronyms and players used long team names, therefore the usage of acronyms doesn't make sense.
If you read this would you know which teams are playing? (if you knew what teams played the tournament in the first place)
![image](https://user-images.githubusercontent.com/24534022/161037723-0831c85f-f207-4811-a710-86fd71215357.png)
This question also ties to #17540

**Reply by peppy** [reply]
Given the limited space, what would you expect to have displayed? Most similar cases I've seen, even outside of osu!, will make acronyms for the teams. Based on their team name, to give a quick identifier to track the team by.

**Reply by 0x84f** [reply]
What I would expect is for the space to be made for the full name to be shown (within reason I guess). The acronyms make it extremely limiting to convey the team name with an identifier that's understandable.

**Reply by peppy** [reply]
I'm willing to entertain the idea of making these panels expand for larger names, but I'd want to see more support from other tournament organisers before putting in the effort. It's going to take a but of restructuring to get things working.

If anyone else comes across this issue as a blocker/required improvement for your tournament, please upvote it or provide more examples of use cases.
